controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -277,37 +280,20 @@ func (r *DataScienceClusterReconciler) ReconcileComponent(
 
 	r.Log.Info("Starting reconciliation of component: " + componentName)
 
-	enabled := component.GetManagementState(instance) == operatorv1.Managed
 	componentCR := component.NewCRObject(instance)
 	err := r.apply(ctx, instance, componentCR)
 	if err != nil {
-		return instance, err
-	}
-
-	_, isExistStatus := instance.Status.InstalledComponents[componentName]
-	if !isExistStatus {
-		message := "Component is disabled"
-		if enabled {
-			message = "Component is enabled"
-		}
-		var err error
-		instance, err = status.UpdateWithRetry(ctx, r.Client, instance, func(saved *dscv1.DataScienceCluster) {
-			status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileInit, message, corev1.ConditionUnknown)
-		})
-		if err != nil {
-			return instance, fmt.Errorf("failed to update DataScienceCluster conditions before first time reconciling %s: %w", componentName, err)
-		}
-	}
-
-	if err != nil {
-		r.Log.Error(err, "Failed to reconcile component: "+componentName)
-		instance = r.reportError(err, instance, fmt.Sprintf("failed to reconcile %s on DataScienceCluster", componentName))
+		r.Log.Error(err, "Failed to create component CR: "+componentName)
+		instance = r.reportError(err, instance, fmt.Sprintf("failed to create %s by DSC", componentName))
 		instance, _ = status.UpdateWithRetry(ctx, r.Client, instance, func(saved *dscv1.DataScienceCluster) {
-			status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileFailed, fmt.Sprintf("Component reconciliation failed: %v", err), corev1.ConditionFalse)
+			status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileFailed, fmt.Sprintf("Component creation failed: %v", err), corev1.ConditionFalse)
```

Comment:
nit: this method does not only creates a component, but would also update it so I think `reconciliation` is more appropriate

Reply:
correct. let me revert it

Reply:
updated